### PR TITLE
ResORR plugin and minor fix for forecasting

### PR DESCRIPTION
**Pull request by SarathUW**
Commit 1
    Fixed runtime issues with forecasting plugin
       -  rat fails if PLUGINS section is not present in config.
       -  removed unused import cfgrib statement
Commit 2
     Adding the ResORR plugin to RAT

### CODE REVIEW

File: src/rat/plugins/forecasting.py
Comment (SanchitMinocha) [inline src/rat/plugins/forecasting.py:1]: Thanks for fixing it.

File: src/rat/plugins/resorr/runResorr.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+import networkx as nx
+import os
+
+from logging import getLogger
+from rat.utils.logging import LOG_NAME, LOG_LEVEL1_NAME
+from rat.utils.utils import create_directory
+from rat.plugins.resorr.data_prep import generate_network
+from resorr.data_prep import generate_forcings_from_rat, generate_network
+from resorr.network import ReservoirNetwork
+from tqdm.notebook import tqdm
```
Comment (SanchitMinocha) [inline src/rat/plugins/resorr/runResorr.py:17]: This can be removed.

**Review by SanchitMinocha — CHANGES_REQUESTED** [review]
I have not read the ResORR code in detail. But it looks good, well-documented and error-handling  also seems great.

File: src/rat/plugins/resorr/runResorr.py (hunk 1 above)
Comment (SarathUW) [reply src/rat/plugins/resorr/runResorr.py]: removed. Thanks for checking the code.

**Review by SanchitMinocha — APPROVED** [review]
Great work. Thank you for fixing it and pushing the plugin code.

**Comment by SanchitMinocha** [comment]
Forgot to mention it earlier - @pritamd47 or @SarathUW 
Can anyone of you mention details of how to use ResORR from within RAT as a plugin in the [Plugins section of RAT docs](https://rat-satellitedams.readthedocs.io/en/latest/Plugins/Forecasting/). You can there mention for complete documentation on how ResORR works at [https://resorr.readthedocs.io/en/latest/](https://resorr.readthedocs.io/en/latest/).

**Pull request merged**